### ISSUE TITLE
MQTT connection frequently gets lost on ESP8266

### ISSUE THREAD

**Issue by trombik** [trombik/esp32-homie#18]
```
I (290403) EXAMPLE: Publishing random value
I (293404) EXAMPLE: Publishing random value
I (296405) EXAMPLE: Publishing random value
I (299406) EXAMPLE: Publishing random value
E (300931) TRANS_TCP: tcp_poll_read select error 104, errno = Connection reset by peer, fd = 0
E (300937) MQTT_CLIENT: Read error or end of stream
```

log from `journalctl -u mosquitto`:

```
Dec 18 06:57:53 hab mosquitto[28447]: Socket error on client homie_example, disconnecting.
```

**Comment by trombik** [comment]
`esp-mqtt` in ESP8266 RTOS SDK is not well-maintained.

https://github.com/espressif/ESP8266_RTOS_SDK/commits/master/components/mqtt/esp-mqtt

**Comment by trombik** [comment]
as documented in 7f6a5806002b4bdff9aec60bd1c9b794e9f250c1, `esp-mqtt` in the SDK is not thread-safe.

**Comment by trombik** [comment]
fixed in e8b67211020ead14fb5f4e9df0a804851591b8c1
